### PR TITLE
Add API documentation

### PR DESCRIPTION
**Pull request by nanaya**
- ~~currently done together with tslint jobs~~ now with more jobs but not a big problem because fast :running_man: 
- the included markdown files are documenting objects and websockets
- assets building is currently rather annoying with its own system and all
  - will be migrated to webpack or something
- copied good chunk of styles from slate (or more like undoing modifications by the php conversion)

Setting up:
- set `TRAVIS_GITHUB_TOKEN`
- set `TRAVIS_APIDOC_REPO` if deployed to another repository (`user/repo`)
- set `TRAVIS_APIDOC_APP_URL` to prod APP_URL
- don't forget to make an empty `gh-pages` branch to the target repository (otherwise travis will explode)

### CODE REVIEW

File: docs/source/includes/_structures.md [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+beatmapset_discussion_unlock   | Discussion on beatmap has been unlocked
+beatmapset_disqualify          | Beatmap was disqualified
+beatmapset_love                | Beatmap was promoted to loved
+beatmapset_nominate            | Beatmap was nominated
+beatmapset_qualify             | Beatmap has gained enough nominations and entered the ranking queue
+beatmapset_reset_nominations   | Nomination of beatmap was reset
+channel_message                | Someone sent chat message
+forum_topic_reply              | Someone replied on forum topic
+
+### `beatmapset_discussion_lock`
```
Comment (nekodex) [inline docs/source/includes/_structures.md]: No examples for these notification objects?

Also, might be worth grouping these by `object_type` or something to make it easier to follow... right now it's a wall of... tables?

File: docs/source/includes/_structures.md [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+object_id        | number  |
+source_user_id   | number? |
+is_read          | boolean |
+details          | object  | `message_id` of last known message (only returned in presence responses)
+
+### Event Names
+
+Name                           | Description
+------------------------------ | -----------------------------------------------------
+beatmapset_discussion_lock     | Discussion on beatmap has been locked
```
Comment (nekodex) [inline docs/source/includes/_structures.md]: is it possible to link these to the relevant sections?

File: app/Http/Controllers/NotificationsController.php [hunk 3]
```diff
@@ -76,6 +124,23 @@ public function index()
...
+     *
+     * ---
+     *
+     * ### Response Format
+     *
+     * _empty response_
+     *
+     * @authenticated
+     *
+     * @bodyParam ids[] integer[] required `id` of notifications to be marked as read.
```
[the remark this answers is not in the document]
Comment (nanaya) [reply app/Http/Controllers/NotificationsController.php]: Dug a bit and found this mpociot/laravel-apidoc-generator/issues/424

File: docs/source/includes/_structures.md (hunk 1 above)
Comment (nanaya) [reply docs/source/includes/_structures.md]: To be done later™.

File: config/apidoc.php [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+     * This path should be relative to the root of your application.
+     */
+    'output' => 'docs',
+
+    /*
+     * The router to be used (Laravel or Dingo).
+     */
+    'router' => 'laravel',
+
+    'faker_seed' => 0,
```
Comment (nekodex) [inline config/apidoc.php]: Yeah, so the seed can't be 0... cuz php:
[mpociot/laravel-apidoc-generator/blob/master/src/Tools/Generator.php#L316](https://github.com/mpociot/laravel-apidoc-generator/blob/master/src/Tools/Generator.php#L316)

**Pull request merged**